src/Examples/Flowsheets/power_generation/rsofc/cpu.py

```diff
@@ -0,0 +1,720 @@
+        )
+        def mole_frac_comp_water_eqn(b, t):
+            return 0 == 1 - sum(
+                b.water_mole_frac_comp[t, c] for c in self.component_list
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Vent stream: component mole flow equation",
+        )
+        def mole_frac_comp_vent_eqn(b, t):
+            return 0 == 1 - sum(
+                b.vent_mole_frac_comp[t, c] for c in self.component_list
+            )
+
+        # Component mole flow = total flow_mol * mole_frac_comp
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.component_list,
+            doc="Inlet component mole flow eqn [mol/s]",
+        )
+        def flow_mol_comp_inlet_eqn(b, t, c):
+            return (
+                b.inlet_flow_mol_comp[t, c]
+                == b.inlet_flow_mol[t] * b.inlet_mole_frac_comp[t, c]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.component_list,
+            doc="PureCO2 component mole flow eqn [mol/s]",
+        )
+        def flow_mol_comp_pureco2_eqn(b, t, c):
+            return (
+                b.pureco2_flow_mol_comp[t, c]
+                == b.pureco2_flow_mol[t] * b.pureco2_mole_frac_comp[t, c]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.component_list,
+            doc="Water component mole flow eqn [mol/s]",
+        )
+        def flow_mol_comp_water_eqn(b, t, c):
+            return (
+                b.water_flow_mol_comp[t, c]
+                == b.water_flow_mol[t] * b.water_mole_frac_comp[t, c]
+            )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.component_list,
+            doc="Vent component mole flow eqn [mol/s]",
+        )
+        def flow_mol_comp_vent_eqn(b, t, c):
+            return (
+                b.vent_flow_mol_comp[t, c]
+                == b.vent_flow_mol[t] * b.vent_mole_frac_comp[t, c]
+            )
+
+        # Total component balance across the unit (i. e. inlet and outlets)
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.component_list,
+            doc="component material balance equation",
+        )
+        def component_material_balance_eqn(b, t, c):
+            return (
+                0
+                == self.inlet_flow_mol_comp[t, c]
+                - self.pureco2_flow_mol_comp[t, c]
+                - self.water_flow_mol_comp[t, c]
+                - self.vent_flow_mol_comp[t, c]
+            )
+
+    def add_surrogates(self):
+        """ This section is to add the surrogate models"""
+
+        # Compressure heat duty
+        @self.Expression(
+            self.flowsheet().config.time, doc="Compressor train heat duty [J/s]"
+        )
+        def heat_duty(b, t):
+            return sm.heat_duty_fun.f(
```

Comment:
It is probably too much work to change now, but it would be really nice if this had been done through the standard surrogate interface.